vpn/tls_test.go

```diff
@@ -480,10 +487,157 @@ func Test_tlsHandshake(t *testing.T) {
 		t.Errorf("tlsHandshake() error = %v, wantErr %v", err, wantErr)
 		return
 	}
+
+	// we bubble up any error coming from the factory
+	tlsFactoryFn = errorRaisingTLSFactory
```

Comment:
If this is a global variable, shouldn't you restore it to its original value in all code paths before leaving this func?

Reply:
I am changing all the occurrences of a global var replace in tests to: 
``
orig := global
global = dummy
defer func() {
  global = orig
}()
```
unless there's a more idiomatic way of doing this.

Reply:
That sounds good! In the future we could refactor to do some form of dependency injection, but I don't see this as a priority right now since that pattern looks good and refactoring would cause some code churn!